src/network.js

```diff
@@ -17,78 +16,80 @@ class Network {
     this.libp2p = libp2p
     this.bitswap = bitswap
-    this.b100Only = options.b100Only || false
+    this.protocols = [BITSWAP100]
+    if (!options.b100Only) {
+      // Latest bitswap first
+      this.protocols.unshift(BITSWAP110)
+    }
 
     this._stats = stats
     this._running = false
-  }
 
-  start () {
-    this._running = true
     // bind event listeners
     this._onPeerConnect = this._onPeerConnect.bind(this)
     this._onPeerDisconnect = this._onPeerDisconnect.bind(this)
-
     this._onConnection = this._onConnection.bind(this)
-    this.libp2p.handle(BITSWAP100, this._onConnection)
-    if (!this.b100Only) { this.libp2p.handle(BITSWAP110, this._onConnection) }
+  }
+
+  start () {
+    this._running = true
+    this.libp2p.handle(this.protocols, this._onConnection)
 
     this.libp2p.on('peer:connect', this._onPeerConnect)
     this.libp2p.on('peer:disconnect', this._onPeerDisconnect)
 
     // All existing connections are like new ones for us
-    this.libp2p.peerBook
-      .getAllArray()
-      .filter((peer) => peer.isConnected())
-      .forEach((peer) => this._onPeerConnect((peer)))
+    for (const peer of this.libp2p.peerStore.peers.values()) {
+      if (this.libp2p.registrar.getConnection(peer)) {
+        this._onPeerConnect(peer)
+      }
+    }
   }
 
   stop () {
     this._running = false
 
-    this.libp2p.unhandle(BITSWAP100)
-    if (!this.b100Only) { this.libp2p.unhandle(BITSWAP110) }
+    // Unhandle both, libp2p doesn't care if it's not already handled
+    this.libp2p.unhandle(this.protocols)
 
     this.libp2p.removeListener('peer:connect', this._onPeerConnect)
     this.libp2p.removeListener('peer:disconnect', this._onPeerDisconnect)
   }
 
-  // Handles both types of bitswap messgages
-  _onConnection (protocol, conn) {
+  /**
+   * Handles both types of incoming bitswap messages
+   * @private
+   * @param {object} param0
+   * @param {string} param0.protocol The protocol the stream is running
+   * @param {Stream} param0.stream A duplex iterable stream
+   * @param {Connection} param0.connection A libp2p Connection
+   * @returns {void}
+   */
+  _onConnection ({ protocol, stream, connection }) {
     if (!this._running) { return }
-    this._log('incomming new bitswap connection: %s', protocol)
+    this._log('incoming new bitswap %s connection from %s', protocol, connection.remotePeer.toB58String())
 
-    pull(
-      conn,
+    pipe(
+      stream,
       lp.decode(),
-      pull.asyncMap((data, cb) => callbackify(Message.deserialize)(data, cb)),
-      pull.asyncMap((msg, cb) => {
-        conn.getPeerInfo((err, peerInfo) => {
-          if (err) {
-            return cb(err)
+      async (source) => {
+        for await (const data of source) {
+          try {
+            const message = await Message.deserialize(data.slice())
+            this.bitswap._receiveMessage(connection.remotePeer, message)
+          } catch (err) {
+            this.bitswap._receiveError(err)
```

Comment:
Do we want to continue receiving or return?

Reply:
To be the same as the existing code it should end iteration, you're right. I added a `break` to exit iteration to keep it consistent with the existing functionality.